src/api.js

```diff
@@ -0,0 +1,119 @@
+import { DELETE, POST, PUT } from './constants.js';
+
+const BASE_URL = 'https://js-todo-list-9ca3a.df.r.appspot.com/api/users';
+
+function getUserURL(userId) {
+  return `${BASE_URL}/${userId}`;
+}
+
+function getTodoListURL(userId) {
+  return `${BASE_URL}/${userId}/items`;
+}
+
+function getTodoItemURL(userId, itemId) {
+  return `${BASE_URL}/${userId}/items/${itemId}`;
+}
+
+function getTodoItemToggleURL(userId, itemId) {
+  return `${BASE_URL}/${userId}/items/${itemId}/toggle`;
+}
+
+function getTodoItemPriorityURL(userId, itemId) {
+  return `${BASE_URL}/${userId}/items/${itemId}/priority`;
+}
+
+export async function getUsersData() {
+  const response = await fetch(BASE_URL);
```

Comment:
네트워크 통신을 처리할 때 try~catch등으로 에러상황을 처리할 수 있으면 더 좋은 코드가 될 것 같습니다!

그리고 fetch API는 response에서 ok로 성공/실패를 확인할 수 있으니 그 점을 추가하면 더 좋을 것 같아요ㅋㅋ

```suggestion
 try {
   const response = await fetch(BASE_URL);
   if (response.ok) {
       ....
   } else {
       throw new Error();
   }
  } catch (e) {
     console.errro(e);
  }
```

Reply:
try catch를 어떻게 써야할지 잘 몰라서 안 썼는데, 이렇게 쓰면 되는군요~ 코드에 적용해보니 깔끔하고 좋은 것 같아요

Reply:
try catch를 어떻게 써야할지 잘 몰라서 안 썼는데, 이렇게 쓰면 되는군요~ 코드에 적용해보니 깔끔하고 좋은 것 같아요